x-pack/plugins/stack_connectors/server/connector_types/inference/index.ts

```diff
@@ -0,0 +1,182 @@
+    );
+  } catch (e) {
+    logger.error(
+      `Failed to delete inference endpoint for task type "${taskType}" and inference id ${inferenceId}. Error: ${e.message}`
+    );
+    throw e;
+  }
+};
+
+export const getConnectorType = (): SubActionConnectorType<Config, Secrets> => ({
+  id: INFERENCE_CONNECTOR_ID,
+  name: INFERENCE_CONNECTOR_TITLE,
+  getService: (params) => new InferenceConnector(params),
+  schema: {
+    config: ConfigSchema,
+    secrets: SecretsSchema,
+  },
+  validators: [{ type: ValidatorType.CONFIG, validator: configValidator }],
+  supportedFeatureIds: [
+    GenerativeAIForSecurityConnectorFeatureId,
+    GenerativeAIForSearchPlaygroundConnectorFeatureId,
+    GenerativeAIForObservabilityConnectorFeatureId,
+  ],
+  minimumLicenseRequired: 'enterprise' as const,
+  preSaveHook: async ({ config, secrets, logger, services, isUpdate }) => {
+    const esClient = services.scopedClusterClient.asCurrentUser;
+    try {
+      const taskSettings = config?.taskTypeConfig
+        ? {
+            ...unflattenObject(config?.taskTypeConfig),
+          }
+        : {};
+      const serviceSettings = {
+        ...unflattenObject(config?.providerConfig ?? {}),
+        ...unflattenObject(secrets?.providerSecrets ?? {}),
+      };
+
+      let inferenceExists = false;
+      try {
+        await esClient?.inference.get({
+          inference_id: config?.inferenceId,
+          task_type: config?.taskType as InferenceTaskType,
+        });
+        inferenceExists = true;
+      } catch (e) {
+        /* throws error if inference endpoint by id does not exist */
+      }
+      if (!isUpdate && inferenceExists) {
+        throw new Error(
+          `Inference with id ${config?.inferenceId} and task type ${config?.taskType} already exists.`
+        );
+      }
+
+      if (isUpdate && inferenceExists && config && config.provider) {
+        // TODO: replace, when update API for inference endpoint exists
+        await deleteInferenceEndpoint(
+          config.inferenceId,
+          config.taskType as InferenceTaskType,
+          logger,
+          esClient
+        );
+      }
+
+      await esClient?.inference.put({
+        inference_id: config?.inferenceId ?? '',
+        task_type: config?.taskType as InferenceTaskType,
+        inference_config: {
+          service: config!.provider,
+          service_settings: serviceSettings,
+          task_settings: taskSettings,
+        },
+      });
+      logger.info(
+        `Inference endpoint for task type "${config?.taskType}" and inference id ${
+          config?.inferenceId
+        } was successfuly ${isUpdate ? 'update' : 'create'}`
+      );
+    } catch (e) {
+      logger.error(
+        `Failed to ${isUpdate ? 'update' : 'create'} inference endpoint for task type "${
+          config?.taskType
+        }" and inference id ${config?.inferenceId}. Error: ${e.message}`
+      );
+      throw e;
```

Comment:
When creating the connector, if I pass mock data saving fails as expected but there is also a 500 in the console which looks wrong.

<img width="1591" alt="Screenshot 2024-10-08 at 16 16 09" src="https://github.com/user-attachments/assets/e038041d-d83b-4f63-9ace-7b39a5892457">

Reply:
@YulNaumenko will know for sure.

I assume the 500 is from the call the create hook is making to an ES `/inference` end-point.  Perhaps it requires something special on the ES side of the house?  Or the hook needs some additional error checks?